virtual_rainforest/data_variables.toml

```diff
@@ -0,0 +1,385 @@
+
+[[variable]]
+name = "shortwave_in"
+description = "downward shortwave radiation"
+unit = "J m-2"
+source = "external"
+used_by = [ "abiotic"]
+
+[[variable]]
+name = "soil_moisture"
+description = "Soil moisture profile"
+unit = "-"
+source = "abiotic"
+used_by = [ "abiotic", "soil"]
+
+[[variable]]
+name = "soil_respiration"
+description = "Soil respiration"
+unit = "ppm"
+source = "soil"
+used_by = [ "abiotic", "soil"]
+
+[[variable]]
+name = "soil_temperature"
+description = "soil temperature profile"
+unit = "C"
+source = "abiotic"
+used_by = [ "abiotic", "soil"]
+
+[[variable]]
+name = "specific_heat_air"
+description = "specific heat of air"
+unit = "kJ kg-1"
+source = "abiotic"
+used_by = [ "abiotic"]
+
+[[variable]]
+name = "specific_humidity"
+description = "specific humidity of air"
+unit = "g kg-1"
+source = "abiotic"
+used_by = [ "abiotic", "plants"]
+
+[[variable]]
+name = "subsurface_runoff"
+description = "Subsurface runoff"
+unit = "m3 s-1"
+source = "abiotic"
+used_by = [ "abiotic", "soil"]
+
+[[variable]]
+name = "sunshine_fraction"
+description = "fraction of sunshine hours, between 0 and 1"
+unit = "-"
+source = "external "
+used_by = [ "abiotic"]
+
+[[variable]]
+name = "surface_runoff"
+description = "surface runoff"
+unit = "m3 s-1"
+source = "abiotic"
+used_by = [ "abiotic", "soil"]
+
+[[variable]]
+name = "surface_water"
+description = "Searchable surface water"
+unit = "%"
+source = "abiotic"
+used_by = [ "abiotic", "soil"]
+
+[[variable]]
+name = "topofcanopy_radiation"
+description = "Top of canopy downward shortwave radiation"
+unit = "J m-2"
+source = "abiotic"
+used_by = [ "abiotic"]
+
+[[variable]]
+name = "wind_above_canopy"
+description = "wind above canopy"
+unit = "m s-1"
+source = "abiotic"
+used_by = [ "abiotic"]
+
+[[variable]]
+name = "wind_below_canopy"
+description = "wind profile within and below canopy"
+unit = "m s-1"
+source = "abiotic"
+used_by = [ "abiotic"]
+
+[[variable]]
+name = "wind_speed_ref"
+description = "wind speed at reference height (10m)"
+unit = "m s-1"
+source = "external "
+used_by = [ "abiotic"]
+
+[[variable]]
+name = "soil_c_pool_lmwc"
+description = "Size of low molecular weight carbon pool"
+unit = "kg C m-3"
+source = "soil"
+used_by = [ "plants", "soil"]
+
+[[variable]]
+name = "soil_c_pool_maom"
+description = "Size of mineral associated organic matter pool"
+unit = "kg C m-3"
+source = "soil"
+used_by = [ "soil"]
+
+[[variable]]
+name = "soil_c_pool_microbe"
+description = "Size of microbial biomass pool"
+unit = "kg C m-3"
+source = "soil"
+used_by = [ "soil"]
+
+[[variable]]
+name = "bulk_density"
+description = "Bulk density of soil"
+unit = "kg m-3"
+source = "external + soil"
+used_by = [ "abiotic", "soil"]
+
+[[variable]]
+name = "percent_clay"
+description = "Clay as a percentage of total soil"
+unit = "%"
+source = "external"
+used_by = [ "abiotic", "soil"]
+
+[[variable]]
+name = "pH"
+description = "Soil pH"
+unit = "pH"
+source = "external + soil"
```

Comment:
We probably also need to make sure we can express this consistently. "source1 + source2" definitely would be parseable downstream, but we would need to make sure we use it in every case

Reply:
oh yes, there was something similar with an abiotic variables, I wanted to put a comment here but forgot. I think it might be best to split in into two variables?

Reply:
I'm not quite sure what it means when there is external _and_ a model? If it means that soil modifies external pH then it isn't a source, but it could mean that soil provides a default? This could be a list of sources? `source = ["external", "soil"]`

Reply:
I mean I wasn't sure how to notate this one, basically pH will be drawn from data but down the line we might introduce a pH model to modify pH as the simulation proceeds (either based on changes in N or due to disturbance scenarios).

Reply:
This could maybe become "modified_by" rather than "source" as I guess that's what we want to capture here

Reply:
Or "updated_by" to align with the BaseModel terminology. And I guess this should only be a single model, so just one string.

Reply:
There is a difference between variables that come from an external source and others that are updated internally. I think they should have different names, such as `pH_input` or something like that. I used `_ref` for variables at reference height.

Reply:
Wouldn't this basically double the number of variables? At least in the soil module pretty much every variable stored in the `data` object will be initialised using external data, duplicating everything seems like it could get pretty confusing. I guess the problem can be avoided once we store variables over time (rather than replacing everything at each time step) as the initial state will be preserved.

Reply:
Though maybe I'm confused and you are referring to continuous data from external sources, which the soil module doesn't have any of at the moment (I doubt it ever will)

Reply:
Ahhh right I misunderstood then 🙃. Completely agree that we should not be altering continuous input data!